src/components/batch-selection/batch-selection.pw.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { BatchSelectionProps } from "components/batch-selection";
+import BatchSelectionComponent from "./components.test-pw";
+import {
+  checkAccessibility,
+  positionOfElement,
+  checkGoldenOutline,
+} from "../../../playwright/support/helper";
+
+import {
+  batchSelectionCounter,
+  batchSelectionComponent,
+  batchSelectionButtonsByPosition,
+} from "../../../playwright/components/batch-selection";
+
+const BATCH_SELECTION_COLOR = [
+  "dark",
+  "light",
+  "white",
+  "transparent",
+] as const;
+
+test.describe("check BatchSelection component properties", () => {
+  [0, 10, 100].forEach((selectedCount) => {
+    test(`check BatchSelection component ${selectedCount} selected Count`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<BatchSelectionComponent selectedCount={selectedCount} />);
+
+      await expect(batchSelectionCounter(page)).toHaveText(
+        `${selectedCount} selected`
+      );
+    });
+  });
+
+  test("should check hidden BatchSelection", async ({ mount, page }) => {
+    await mount(<BatchSelectionComponent hidden />);
+
+    await expect(batchSelectionComponent(page).getAttribute("hidden"))
+      .toBeTruthy;
+
+    await expect(batchSelectionComponent(page)).toBeVisible();
+  });
+
+  test("should check disabled BatchSelection", async ({ mount, page }) => {
+    await mount(<BatchSelectionComponent disabled />);
+
+    await expect(batchSelectionComponent(page).getAttribute("disabled"))
+      .toBeTruthy;
+  });
+
+  ([
+    [BATCH_SELECTION_COLOR[0], "rgb(0, 50, 76)"],
+    [BATCH_SELECTION_COLOR[1], "rgb(179, 194, 201)"],
+    [BATCH_SELECTION_COLOR[2], "rgb(255, 255, 255)"],
+    [BATCH_SELECTION_COLOR[3], ""],
+  ] as [BatchSelectionProps["colorTheme"], string][]).forEach(
+    ([colorTheme, backgroundColor]) => {
+      test(`check BatchSelection component ${colorTheme} colorTheme and it uses ${backgroundColor} as a background color`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <BatchSelectionComponent colorTheme={colorTheme} selectedCount={0} />
+        );
+
+        if (colorTheme === "transparent") {
+          await expect(batchSelectionComponent(page)).not.toHaveCSS(
+            "background-color",
+            backgroundColor
+          );
+        } else {
+          await expect(batchSelectionComponent(page)).toHaveCSS(
+            "background-color",
+            backgroundColor
+          );
+        }
+      });
+    }
+  );
+});
+
+test.describe("check BatchSelection buttons are focused", () => {
+  (["first", "second", "third"] as const).forEach((index) => {
+    test(`should check BatchSelection ${index} button has expected styling when focusRedesignOptOut is true`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<BatchSelectionComponent selectedCount={1} />, {
```

Comment:
```suggestion
      await mount<HooksConfig>(<BatchSelectionComponent selectedCount={1} />, {
```

Reply:
You'll need to import `HooksConfig` too but VS Code should be able to sort that import out for you 😄